src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -1,234 +1,235 @@
+            string? serviceName,
+            TInput input,
+            Action<DownstreamApiOptions>? downstreamApiOptionsOverride = null,
+            CancellationToken cancellationToken = default) where TOutput : class
+        {
+            DownstreamApiOptions effectiveOptions = MergeOptions(serviceName, downstreamApiOptionsOverride);
+            HttpContent? effectiveInput = SerializeInput(input, effectiveOptions);
+            HttpResponseMessage response = await CallApiInternalAsync(serviceName, effectiveOptions, true,
+                                                                          effectiveInput, null, cancellationToken).ConfigureAwait(false);
+
+            // Only dispose the HttpContent if was created here, not provided by the caller.
+            if (input is not HttpContent)
+            {
+                effectiveInput?.Dispose();
+            }
+
+            return await DeserializeOutput<TOutput>(response, effectiveOptions).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public async Task<TOutput?> CallApiForAppAsync<TOutput>(string serviceName,
+            Action<DownstreamApiOptions>? downstreamApiOptionsOverride = null,
+            CancellationToken cancellationToken = default) where TOutput : class
+        {
+            DownstreamApiOptions effectiveOptions = MergeOptions(serviceName, downstreamApiOptionsOverride);
+            HttpResponseMessage response = await CallApiInternalAsync(serviceName, effectiveOptions, true,
+                                                                          null, null, cancellationToken).ConfigureAwait(false);
+
+            return await DeserializeOutput<TOutput>(response, effectiveOptions).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public async Task<TOutput?> CallApiForUserAsync<TOutput>(
+            string? serviceName,
+            Action<DownstreamApiOptions>? downstreamApiOptionsOverride = null,
+            ClaimsPrincipal? user = null,
+            CancellationToken cancellationToken = default) where TOutput : class
+        {
+            DownstreamApiOptions effectiveOptions = MergeOptions(serviceName, downstreamApiOptionsOverride);
+            HttpResponseMessage response = await CallApiInternalAsync(serviceName, effectiveOptions, false,
+                                                                          null, user, cancellationToken).ConfigureAwait(false);
+            return await DeserializeOutput<TOutput>(response, effectiveOptions).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Merge the options from configuration and override from caller.
+        /// </summary>
+        /// <param name="optionsInstanceName">Named configuration.</param>
+        /// <param name="calledApiOptionsOverride">Delegate to override the configuration.</param>
+        private /* for tests */ DownstreamApiOptions MergeOptions(

```

Comment:
why change the visibility?
This is not needed.
Please revert that. This is used by tests.

Reply:
It is not unit tested :( - as you can see the CI passed.

Reply:
Agree that it's not necessary as part of this PR.

Reply:
Can you please elaborate why I should increase the visibility from private to internal for these methods? They are only used from inside the class. There are no tests actioning them. 

Are there maybe other projects outside Id.Web which use them via InteralVisibleTo?

Reply:
@bgavrilMS/ We have a lot to do in many projects. and I personally don't want us to have noise in the PR that I need to review. I'm not opposed to clean-up as we go, but I'd prefer you add tests if you have energy than changing things that are not strictly necessary.